File: backend/src/main/kotlin/com/example/backend/models/Facturation.kt
package com.example.backend.models

import jakarta.persistence.*
import org.hibernate.annotations.ColumnDefault
import java.time.LocalDateTime

@Entity
@Table(name = "facturation")
class Facturation {
  @Id
  @Column(name = "id", nullable = false)
  var id: Int? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "vente_id")
  var vente: com.example.backend.models.Vente? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "caisse_id")
  var caisse: Caisse? = null

  @Column(name = "typePaiement", length = 100)
  var typePaiement: String? = null

  @Column(name = "montantPercu")
  var montantPercu: Int? = null

  @Column(name = "reste")
  var reste: Int? = null

  @Column(name = "montantTtc")
  var montantTtc: Int? = null

  @Column(name = "dateFacture")
  var dateFacture: LocalDateTime? = null

  @ColumnDefault("0")
  @Column(name = "supprimer")
  var supprimer: Int? = null
}

File: backend_py/pharmaPython/app/models/facturation.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from app.api.v1.db import Base

class Facturation(Base):
  __tablename__ = "facturation"
  __table_args__ = {"extend_existing": True}

  id = Column(Integer, primary_key=True)
  vente_id = Column(Integer, ForeignKey("vente.id"), nullable=False)
  caisse_id = Column(Integer, ForeignKey("caisse.id"), nullable=False)
  vente = relationship("Vente", lazy="joined")
  caisse = relationship("Caisse", lazy="joined")
  type_paiement = Column(String(100))
  montant_percu = Column(Integer)
  reste = Column(Integer)
  montant_ttc = Column(Integer)
  date_facture = Column(DateTime)
  supprimer = Column(Integer, default=0)
